sdk/attestation/azure-security-attestation/tests/test_policy_getset.py

```diff
@@ -27,268 +27,323 @@
+        )
 
-
-    @AttestationPreparer()
-    def test_isolated_get_policy_sgx(self, attestation_isolated_url):
-        attest_client = self.create_admin_client(attestation_isolated_url)
-        policy_response = attest_client.get_policy(AttestationType.SGX_ENCLAVE)
-        print('Shared policy: ', policy_response.value)
-        assert(policy_response.value.startswith('version'))
-        print('Token: ', policy_response.token)
-
-    @AttestationPreparer()
-    def test_aad_get_policy_sgx(self, attestation_aad_url):
+        attestation_type = kwargs.pop("attestation_type")
         attest_client = self.create_admin_client(attestation_aad_url)
-        policy_response = attest_client.get_policy(AttestationType.SGX_ENCLAVE)
-        print('Shared policy: ', policy_response.value)
-        assert(policy_response.value.startswith('version'))
-        print('Token: ', policy_response.token)
+        policy_set_response, _ = attest_client.set_policy(
+            attestation_type, attestation_policy
+        )
+        new_policy = attest_client.get_policy(attestation_type)[0]
+        assert new_policy == attestation_policy
 
-    @AttestationPreparer()
-    def test_aad_set_policy_sgx_unsecured(self, attestation_aad_url):
-        attestation_policy = u"version=1.0; authorizationrules{=> permit();}; issuancerules{};"
-
-        attest_client = self.create_admin_client(attestation_aad_url)
-        policy_set_response = attest_client.set_policy(AttestationType.SGX_ENCLAVE, attestation_policy)
-        policy_get_response = attest_client.get_policy(AttestationType.SGX_ENCLAVE)
-        assert policy_get_response.value == attestation_policy
+        expected_policy = AttestationPolicyToken(attestation_policy)
 
-        expected_policy = AttestationToken(body=StoredAttestationPolicy(attestation_policy))
         hasher = hashes.Hash(hashes.SHA256(), backend=default_backend())
-        hasher.update(expected_policy.serialize().encode('utf-8'))
+        hasher.update(expected_policy.to_jwt_string().encode("utf-8"))
         expected_hash = hasher.finalize()
 
-        assert expected_hash == policy_set_response.value.policy_token_hash
+        assert expected_hash == policy_set_response.policy_token_hash
 
     @AttestationPreparer()
-    def test_aad_reset_policy_sgx_unsecured(self, attestation_aad_url):
+    @AllAttestationTypes
+    def test_aad_reset_policy_unsecured(self, attestation_aad_url, **kwargs):
 
+        attestation_type = kwargs.pop("attestation_type")
         attest_client = self.create_admin_client(attestation_aad_url)
-        policy_set_response = attest_client.reset_policy(AttestationType.SGX_ENCLAVE)
+        policy_set_response, _ = attest_client.reset_policy(attestation_type)
 
-        assert None == policy_set_response.value.policy_token_hash
-        assert policy_set_response.value.policy_resolution == PolicyModification.REMOVED
+        assert None == policy_set_response.policy_token_hash
```

Comment:
nit: this is more typically done like this (you could also do `assert not x`, but that would pass if it's a non-None falsy value)
```suggestion
        assert policy_set_response.policy_token_hash is None
```